### PR TITLE
Use Sphinx Inline Tabs to organise installation per OS

### PR DESCRIPTION
**Pull request by hugovk**
Follow on from https://github.com/python-pillow/Pillow/issues/6755 / https://github.com/python-pillow/Pillow/pull/6756

Changes proposed in this pull request:

 * Use [Sphinx Inline Tabs](https://sphinx-inline-tabs.readthedocs.io/) to group the installation instructions by operating system
 * This helps hide irrelevant information on long, detailed page
 * If, for example, you click the macOS tab in the first tab group, it also opens the macOS tab in the second group
 * Because the subheaders are gone, I added in `.. _Linux Installation:` etc. to avoid linkrot
 * I re-ordered them to go: Linux, macOS, Windows, [Windows variant,] FreeBSD, [Android]
   * My thinking was to put the main three first, in order of "likelihood" to needs to follow these instructions, then the others afterwards, but happy to change it

image

image

Plus some other cleanup:

* Like our top-level Makefile, inline the `.PHONY` targets to help avoid omissions in the docs `Makefile` (and `texinfo`, `info`, `livehtml` and `serve` were missing)
* Add `serve` and `livehtml` to `make -C docs help`
* EditorConfig: three-space indents for RST

# Demo

https://pillow--6764.org.readthedocs.build/en/6764/installation.html

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -39,49 +40,57 @@ help:
...
 
+.PHONY: htmlhelp
 htmlhelp:
 	$(MAKE) install-sphinx
 	$(SPHINXBUILD) -b htmlhelp $(ALLSPHINXOPTS) $(BUILDDIR)/htmlhelp
 	@echo
 	@echo "Build finished; now you can run HTML Help Workshop with the" \
 	      ".hhp project file in $(BUILDDIR)/htmlhelp."
 
+.PHONY:  qthelp
```
Comment (radarhere) [inline docs/Makefile:93]:
```suggestion
.PHONY: qthelp
```

File: .editorconfig [hunk 2]
```diff
@@ -13,6 +13,10 @@ indent_style = space
 
 trim_trailing_whitespace = true
 
+[*.rst]
+# Three-space indentation
+indent_size = 3
```
Comment (radarhere) [inline .editorconfig:18]: I was wondering why 3, and then I found
https://devguide.python.org/documentation/style-guide/
> All reST files use an indentation of 3 spaces

Comment (hugovk) [reply .editorconfig:18]: Yeah, it's a bit weird but then so is RST :)

https://stackoverflow.com/q/48311159/724176 says there's no "official" requirement.

Another random Google result:

> In reST, the indentation of a block of lines is often important. The exact number of spaces, which are used to indent a block of text, does not matter. But what does matter, is that all lines of the block are indented with exactly the same number of spaces.
>
> We use the convention of 4 spaces per indenting level (see [Whitespace and indentation](https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/GeneralConventions/CodingGuidelines.html#cgl-indenting)).

https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/BasicRestSyntax.html

The installation page uses 3 spaces. I did a cursory search of the docs and `git grep "  \.\." docs` suggests we're already using 3-space indents, but happy to remove this config if you prefer.

On the other hand, if we are going to use 3-space, this helps the IDE know how many to add when tabbing.

Comment (radarhere) [reply .editorconfig:18]: Opening up the files inside docs/handbook, I don't think we (at least, not consistently) using 3 spaces at the moment. I think that the mix of 3 spaces for RST and 4 for Python indentation is weird.

But given that there's already a variety of style rules that we have to follow when writing documentation, I'm not overly concerned about adding another, and I think that specifying a value is helpful.

**Review by radarhere — APPROVED** [review]

File: .editorconfig (hunk 2 above)
Comment (hugovk) [reply .editorconfig:18]: Shall we prefer 4 over 3? I'm not too bothered which, and let's not bother mass-reindenting the docs.

Comment (aclark4life) [reply .editorconfig:18]: 4!

Comment (hugovk) [reply .editorconfig]: Updated to 4! Let's do a squash merge on this PR to keep the Git history a little neater.

File: .editorconfig [hunk 3]
```diff
@@ -13,6 +13,10 @@ indent_style = space
 
 trim_trailing_whitespace = true
 
+[*.rst]
+# Three-space indentation
```
Comment (radarhere) [inline .editorconfig:17]:
```suggestion
# Four-space indentation
```

**Pull request merged**